cmake/modules/FindAvahi.cmake

```diff
@@ -40,17 +36,17 @@ if(NOT TARGET Avahi::Avahi)
                                     VERSION_VAR AVAHI_VERSION)
 
   if(AVAHI_FOUND)
-    add_library(Avahi::AvahiCommon UNKNOWN IMPORTED)
-    set_target_properties(Avahi::AvahiCommon PROPERTIES
-                                             IMPORTED_LOCATION "${AVAHI_COMMON_LIBRARY}"
-                                             INTERFACE_INCLUDE_DIRECTORIES "${AVAHI_COMMON_INCLUDE_DIR}"
-                                             INTERFACE_COMPILE_DEFINITIONS "HAS_AVAHI=1;HAS_ZEROCONF=1")
-    add_library(Avahi::Avahi UNKNOWN IMPORTED)
-    set_target_properties(Avahi::Avahi PROPERTIES
-                                       IMPORTED_LOCATION "${AVAHI_CLIENT_LIBRARY}"
-                                       INTERFACE_INCLUDE_DIRECTORIES "${AVAHI_CLIENT_INCLUDE_DIR}"
-                                       INTERFACE_COMPILE_DEFINITIONS "HAS_AVAHI=1;HAS_ZEROCONF=1"
-                                       INTERFACE_LINK_LIBRARIES Avahi::AvahiCommon)
-    set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP Avahi::Avahi)
+    add_library(${APP_NAME_LC}::AvahiCommon UNKNOWN IMPORTED)
+    set_target_properties(${APP_NAME_LC}::AvahiCommon PROPERTIES
+                                                      IMPORTED_LOCATION "${AVAHI_COMMON_LIBRARY}"
+                                                      INTERFACE_INCLUDE_DIRECTORIES "${AVAHI_COMMON_INCLUDE_DIR}"
+                                                      INTERFACE_COMPILE_DEFINITIONS "HAS_AVAHI;HAS_ZEROCONF")
+    add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} UNKNOWN IMPORTED)
+    set_target_properties(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} PROPERTIES
+                                                                     IMPORTED_LOCATION "${AVAHI_CLIENT_LIBRARY}"
+                                                                     INTERFACE_INCLUDE_DIRECTORIES "${AVAHI_CLIENT_INCLUDE_DIR}"
+                                                                     INTERFACE_COMPILE_DEFINITIONS "HAS_AVAHI;HAS_ZEROCONF"
+                                                                     INTERFACE_LINK_LIBRARIES Avahi::AvahiCommon)
```

Comment:
@fuzzard: I think
`INTERFACE_LINK_LIBRARIES Avahi::AvahiCommon` should now be
`INTERFACE_LINK_LIBRARIES ${APP_NAME_LC}::AvahiCommon` ?

Because I get
```
CMake Error at cmake/modules/FindAvahi.cmake:45 (set_target_properties):
  The link interface of target "kodi::Avahi" contains:

    Avahi::AvahiCommon

  but the target was not found.  Possible reasons include:
```